src/core/pipelines/core_pipeline.ts

```diff
@@ -244,35 +258,37 @@ export default function createPipeline<T, U, V>(
       const request$ = downloadingBackoff<ILoaderEvent<U>>(
         tryCatch<T, ILoaderEvent<U>>(loader as any, loaderArgument),
         backoffOptions
-      )
+      ).pipe(
 
-        .catch((error : Error) : Observable<never> => {
+        catchError((error : Error) : Observable<never> => {
           throw errorSelector("PIPELINE_LOAD_ERROR", error);
-        })
+        }),
 
-        .do((arg) => {
+        tap((arg) => {
           if (arg.type === "response" && cache) {
             cache.add(loaderArgument, arg.value);
           }
-        });
+        })
+      );
 
-      return Observable.of({
-        type: "request" as "request",
-        value: loaderArgument,
-      }).concat(request$);
+      return observableConcat(
+        observableOf({ type: "request" as "request", value: loaderArgument }),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=236&issues=AWO8St0R4427HkKVbpWJ&open=AWO8St0R4427HkKVbpWJ)

[ISSUE_KEY]: AWO8St0R4427HkKVbpWJ